app/app/src/main/java/org/schabi/newpipe/crashreporttomarkdown/MainActivity.java

```diff
@@ -0,0 +1,179 @@
+                    clipboardManager.setPrimaryClip(clipData);
+                    Toast.makeText(getApplicationContext(), getString(R.string.copied_to_clipboard), Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        preview.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String text = humanReadableEditText.getText().toString();
+                if (text.isEmpty()) {
+                    Toast.makeText(getApplicationContext(), R.string.empty_text, Toast.LENGTH_SHORT).show();
+                } else if (text.equals(getString(R.string.error_while_parsing))) {
+                    Toast.makeText(getApplicationContext(), R.string.parsing_error_message, Toast.LENGTH_SHORT).show();
+                } else {
+                    Intent intent = new Intent(getApplicationContext(), PreviewActivity.class);
+                    intent.putExtra(getString(R.string.markdown_text_key), text);
+                    startActivity(intent);
+                }
+            }
+        });
+
+    }
+
+    private void checkIntents() {
+        // Get intent, action and MIME type
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        String type = intent.getType();
+
+        if (Intent.ACTION_SEND.equals(action) && type != null && (type.equals("text/plain") || type.equals("text/json"))) {
+            handleSendText(intent); // Handle text being sent
+        }
+    }
+
+    private void handleSendText(Intent intent) {
+        String sharedText = intent.getStringExtra(Intent.EXTRA_TEXT);
+        if (sharedText != null) {
+            jsonEditText.setText(sharedText);
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        // Inflate the menu; this adds items to the action bar if it is present.
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+
+        switch (item.getItemId()) {
+            case (R.id.clear):
+                if (jsonEditText.getText().toString().isEmpty() && humanReadableEditText.getText().toString().isEmpty()) {
+                    Toast.makeText(getApplicationContext(), R.string.already_empty, Toast.LENGTH_SHORT).show();
+                } else {
+                    jsonEditText.setText("");
+                    humanReadableEditText.setText("");
+                    Toast.makeText(getApplicationContext(), R.string.text_cleared, Toast.LENGTH_SHORT).show();
+                }
+                break;
+            case (R.id.theme):
+                changeTheme();
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    private void changeTheme() {
+        SharedPreferences settings = getApplicationContext().getSharedPreferences(THEME, 0);
```

Comment:
The theme should be checked and eventually updated every time the Activity is resumed (i.e. every time an activity above it is closed). Otherwise if the theme is changed in a stacked Activity and the user goes back to this Activity, the theme wouldn't change. Now the theme-changer button is in the bottom activity, so this is not an issue, but if in the future we add a settings activity, this would become buggy since the bottom activity is not recreated. I suggest creating a base ThemedActivity class to be extended by all Activities, and adding a check in the onResume method that calls recreate() if the theme changed. For example, take a look at this: https://github.com/Stypox/dicio-android/blob/master/app/src/main/java/com/dicio/dicio_android/util/ThemedActivity.java

Reply:
What I did with another test app is a Base activity which would do about the same thing for theme + do other thing (check language too), all activities extends this activity so that there is no need to do the thing I did in NewPipe, ie assureCorrectAppLanguage(context).

But this time I searched again, and found out the Android documentation about dark theme:
https://developer.android.com/guide/topics/ui/look-and-feel/darktheme?hl=fr

Read this:
> Note: Starting with AppCompat v1.1.0, setDefaultNightMode() automatically recreates any started activities.

If I understood this note, there is no need for checking theme everytime onResume